framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -2151,24 +2016,133 @@ public boolean shouldSuppressWarnings(@Nullable Element elt, String errKey) {
+    /**
+     * Determines whether an error (whose message key is {@code messageKey}) should be suppressed.
+     * It is suppressed if any of the given SuppressWarnings strings suppresses it.
+     *
+     * <p>A SuppressWarnings string may be of the following pattern:
+     *
+     * <ol>
+     *   <li>{@code "prefix"}, where prefix is a SuppressWarnings prefix, as specified by {@link
+     *       #getSuppressWarningsPrefixes()}. For example, {@code "nullness"} and {@code
+     *       "initialization"} for the Nullness Checker, {@code "regex"} for the Regex Checker.
+     *   <li>{@code "partial-message-key"}, where partial-message-key is a prefix or suffix of the
+     *       message key that it may suppress. So "generic.argument" would suppress any errors whose
+     *       message key contains "generic.argument".
+     *   <li>{@code "prefix:partial-message-key}, where the prefix and partial-message-key is as
+     *       above. So "nullness:generic.argument", would suppress any errors in the Nullness
+     *       Checker with a message key that contains "generic.argument".
+     * </ol>
+     *
+     * {@code "allcheckers"} is a prefix that suppresses a warning from any checker. {@code "all"}
+     * is a partial-message-key that suppresses a warning with any message key.
+     *
+     * <p>If the {@code -ArequirePrefixInWarningSuppressions} command-line argument was supplied,
+     * then {@code "partial-message-key"} has no effect; {@code "prefix"} and {@code
+     * "prefix:partial-message-key"} are the only SuppressWarnings strings that have an effect.
+     *
+     * @param messageKey the message key of the error the checker is emitting; a lowercase string,
+     *     without any "checkername:" prefix
+     * @param suppressWarningsStrings the SuppressWarnings strings that are in effect. May be null,
+     *     in which case this method returns false.
+     * @return true if an element of {@code suppressWarningsStrings} suppresses the error
+     */
+    private boolean shouldSuppressWarningsStringSuppressKey(
+            String messageKey, String @Nullable ... suppressWarningsStrings) {
+        Set<String> prefixes = this.getSuppressWarningsPrefixes();
+        return shouldSuppressWarningsStringSuppressKey(
+                prefixes, messageKey, suppressWarningsStrings);
+    }
+
+    /**
+     * See {@link #shouldSuppressWarningsStringSuppressKey(String, String...)}
+     *
+     * @param prefixes the SuppressWarnings prefixes used by this checker
+     * @param messageKey the message key of the error the checker is emitting; a lowercase string,
+     *     without any "checkername:" prefix
+     * @param suppressWarningsStrings the SuppressWarnings strings that are in effect. May be null,
+     *     in which case this method returns false.
+     * @return true if one of the {@code suppressWarningsStrings} suppresses the error
+     */
+    private boolean shouldSuppressWarningsStringSuppressKey(
```

Comment:
How about renaming this to just "shouldSuppress" or to some other shorter name?  I find the current name a bit unwieldy, and it doesn't seem to convey more information than the shorter one.  Also, this method is not really about just keys as the method name implies.
I would rename the other overload as well.